### DISCUSSION TITLE
Invalid hook call. Hooks can only be called inside of the body of a function component.

### DISCUSSION THREAD

**Discussion by codingwithashu** [TanStack/query#6047] [General]
I am using Shadcn UI, 
I have created Dialog as a component and passed the form as children (props) so whenever button submit getting error like 
"Invalid hook call. Hooks can only be called inside of the body of a function component."

This is Dialogbox 

![image](https://github.com/TanStack/query/assets/111554315/97bf477e-30da-4bdd-9be5-fa25ad72d386)

this form with button with **useQuery** --
![image](https://github.com/TanStack/query/assets/111554315/758e30eb-d36a-4b4a-aad4-ced1f207355a)

This component 

![image](https://github.com/TanStack/query/assets/111554315/572c4aaf-f711-42ff-93df-0d65456b9e85)

**Comment by TkDodo** [comment]
you're calling `DialogForm` as a function:

```
children={DialogForm(...)}
```

to have hooks work properly, you have to "call" them with jsx:

```
children={<DialogForm ... />}
```

see: https://kentcdodds.com/blog/dont-call-a-react-function-component

**Reply by codingwithashu** [reply]
CancelEmpLeave function uses useQuery, how to call this on button click ?
whenever I call this error comes like "Invalid hook call. Hooks can only be called inside of the body of a function component."
